Samples/HelloWorld/OrleansHost/OrleansHostWrapper.cs

```diff
@@ -82,88 +80,83 @@ public int Stop()
             return 0;
         }
 
-        private SiloArgs ParseArguments(string[] args)
+        private class SiloArgs
         {
-            string deploymentId = null;
-            string siloName = null;
+            public SiloArgs(string siloName, string deploymentId)
+            {
+                this.DeploymentId = deploymentId;
+                this.SiloName = siloName;
+            }
 
-            for (int i = 0; i < args.Length; i++)
+            public static SiloArgs ParseArguments(string[] args)
             {
-                string arg = args[i];
-                if (arg.StartsWith("-") || arg.StartsWith("/"))
+                string deploymentId = null;
+                string siloName = null;
+
+                for (int i = 0; i < args.Length; i++)
                 {
-                    switch (arg.ToLowerInvariant())
+                    string arg = args[i];
+                    if (arg.StartsWith("-") || arg.StartsWith("/"))
                     {
-                        case "/?":
-                        case "/help":
-                        case "-?":
-                        case "-help":
-                            // Query usage help
-                            PrintUsage();
-                            return null;
-                        default:
-                            Console.WriteLine("Bad command line arguments supplied: " + arg);
-                            return null;
+                        switch (arg.ToLowerInvariant())
+                        {
+                            case "/?":
+                            case "/help":
+                            case "-?":
+                            case "-help":
+                                // Query usage help
+                                PrintUsage();
+                                return null;
+                            default:
+                                Console.WriteLine($"Bad command line arguments supplied: {arg}");
+                                return null;
```

Comment:
I guess we should do the opposite and not call PrintUsage() in help, as returning null will already force printing the usage.